src/ufuncs.c

```diff
@@ -223,6 +261,40 @@ static void Y_Y_func(char **args, npy_intp *dimensions,
 }
 static PyUFuncGenericFunction Y_Y_funcs[1] = {&Y_Y_func};
 
+
+/* Define the geom -> prepared geom functions (Y_Yp) */
+static void *prepare_data[1] = {GEOSPrepare_r};
+typedef void *FuncGEOS_Y_Yp(void *context, void *a);
+static char Y_Yp_dtypes[2] = {NPY_OBJECT, NPY_OBJECT};
+static void Y_Yp_func(char **args, npy_intp *dimensions,
+                      npy_intp *steps, void *data)
+{
+    FuncGEOS_Y_Yp *func = (FuncGEOS_Y_Yp *)data;
+    void *context_handle = geos_context[0];
+    GEOSGeometry *in1;
+    GEOSPreparedGeometry *ret_ptr;
+
+    UNARY_LOOP {
+        /* get the geometry: return on error */
+        if (!get_geom(*(GeometryObject **)ip1, &in1)) { return; }
+
+        if (in1 == NULL) {
+            ret_ptr = NULL;
+        } else {
+            ret_ptr = func(context_handle, in1);
+            /* trust that GEOS calls HandleGEOSError on error */
+        }
+    GeometryObject *geom_obj = *(GeometryObject **)ip1;
+    geom_obj->ptr_prepared = ret_ptr;
+    /* TODO check if nout can be 0 for ufuncs (because in principle we don't need to return something here) */
+    PyObject **out = (PyObject **)op1;
+    Py_XDECREF(*out);
+    *out = geom_obj;
```

Comment:
I copy pasted some lines of code here to get it working, but in principle we don't need to return anything here (the geoms get modified in place). But I need to check if ufuncs can have zero return values.

(this is the part that can still give segfault after a while, as I suppose the return value here is not properly set up for GC)